google/cloud/internal/backoff_policy.cc

```diff
@@ -42,14 +42,18 @@ std::chrono::milliseconds ExponentialBackoffPolicy::OnCompletion() {
     generator_ = google::cloud::internal::MakeDefaultPRNG();
   }
   std::uniform_int_distribution<microseconds::rep> rng_distribution(
-      current_delay_range_.count() / 2, current_delay_range_.count());
+      current_delay_start_.count(), current_delay_end_.count());
   // Randomized sleep period because it is possible that after some time all
   // client have same sleep period if we use only exponential backoff policy.
   auto delay = microseconds(rng_distribution(*generator_));
-  current_delay_range_ = microseconds(static_cast<microseconds::rep>(
-      static_cast<double>(current_delay_range_.count()) * scaling_));
-  if (current_delay_range_ >= maximum_delay_) {
-    current_delay_range_ = maximum_delay_;
+
+  current_delay_start_ = current_delay_start_ == current_delay_end_
+                             ? initial_delay_
```

Comment:
Q.  Is the specification really to start again at the initial delay?  I don't know, but it seems strange.

Reply:
Updated the specification. It isn't to start again at the current delay, but if we have an edge case like

```
  ExponentialBackoffPolicy tested(ms(10), ms(12), 2.0);

  auto delay = tested.OnCompletion();
  EXPECT_LE(ms(10), delay);
  EXPECT_GE(ms(12), delay);
  delay = tested.OnCompletion();
  EXPECT_LE(ms(10), delay);
  EXPECT_GE(ms(12), delay);
```

We don't want it to the delay range to keep growing, we want to the policy to maintain the max and min values

Reply:
I'm all for keeping values with the min:max range, but starting back at the initial delay after the start hits the end seems contrary to "backoff".